src/webpack-core/ContainerPlugin.js

```diff
@@ -0,0 +1,198 @@
+const AsyncDependenciesBlock = require("webpack/lib/AsyncDependenciesBlock");
+const RuntimeGlobals = require("webpack/lib/RuntimeGlobals");
+const ModuleFactory = require("webpack/lib/ModuleFactory");
+const { ConcatSource } = require("webpack-sources");
+
+const PLUGIN_NAME = "ContainerPlugin";
+
+class ContainerExposedDependency extends ModuleDependency {
+  constructor(name, request) {
+    super(request);
+    this._name = name;
+  }
+
+  get exposedName() {
+    return this._name;
+  }
+
+  getResourceIdentifier() {
+    return `exposed dependency ${this._name}`;
+  }
+}
+
+class ContainerEntryDependency extends Dependency {
+  constructor(dependencies, name) {
+    super();
+    this.exposedDependencies = dependencies;
+    this.optional = true;
+    this.loc = { name };
+  }
+}
+
+const CONTAINER_ENTRY_MODULE_TYPES = new Set(["javascript"]);
+
+class ContainerEntryModule extends Module {
+  constructor(dependency) {
+    super("javascript/dynamic", null);
+    this.expose = dependency.exposedDependencies;
+  }
+
+  getSourceTypes() {
+    return CONTAINER_ENTRY_MODULE_TYPES;
+  }
+
+  identifier() {
+    return `container entry ${JSON.stringify(
+      this.expose.map(item => item.getResourceIdentifier())
+    )}`;
+  }
+
+  readableIdentifier() {
+    return `container entry`;
+  }
+
+  needBuild(context, callback) {
+    return callback(null, !this.buildMeta);
+  }
+
+  build(options, compilation, resolver, fs, callback) {
+    this.buildMeta = {};
+    this.buildInfo = {
+      strict: true
+    };
+
+    this.clearDependenciesAndBlocks();
+
+    for (const dep of this.expose) {
+      const block = new AsyncDependenciesBlock(
+        undefined,
+        dep.loc,
+        dep.userRequest
+      );
+      block.addDependency(dep);
+      this.addBlock(block);
+    }
+
+    callback();
+  }
+
+  codeGeneration({ moduleGraph, chunkGraph, runtimeTemplate }) {
+    const sources = new Map();
+    const runtimeRequirements = new Set([
+      RuntimeGlobals.exports,
+      RuntimeGlobals.require
+    ]);
+
+    const source = new ConcatSource();
+    sources.set("javascript", source);
+
+    const getters = [];
+
+    for (const block of this.blocks) {
+      const {
+        dependencies: [dep]
+      } = block;
+      const name = dep.exposedName;
+      const mod = moduleGraph.getModule(dep);
+
+      if (!mod) continue; // TODO: Perhaps we log something in non production builds?
+
+      const moduleId = JSON.stringify(mod.id); // TODO: This is temp, `.id` will be gone in v6?
+
+      const require_statement = runtimeTemplate.blockPromise({
+        block,
+        message: `${dep.userRequest}`, // TODO: Should we use the request here?
+        chunkGraph,
+        runtimeRequirements
+      });
+
+      getters.push(
+        `case "${name}":\nreturn ${require_statement}.then(() => ${RuntimeGlobals.require}(${moduleId}));`
+      );
+    }
+
+    source.add(`
+      exports["get"] = function get(module) {
```

Comment:
@sokra is there webpack internals I can use to generate the `__webpack_exports__` block like harmony does?

<img width="491" alt="Screen Shot 2020-02-14 at 23 47 14" src="https://user-images.githubusercontent.com/599459/74536714-5ba46800-4f84-11ea-9c37-fc19a3c493b0.png">